### ISSUE TITLE
PayJoin

### ISSUE THREAD

**Issue by hsjoberg** [hsjoberg/blixt-wallet#1262]

**Comment by DanGould** [comment]
Thanks for bringing up this issue. I see two sane approaches to integrating send support [as you mention on Telegram](https://t.me/blixtwallet/168967)

## 1. TypeScript bindings to WASM Payjoin Dev Kit

Boltz's @michael1011 has expressed support for a WASM bindings library based on PDK's `rust-payjoin`. WASM bindings are special in that they'd run just like any other npm library and would not require a separate process or thread to run like Tor would. Disclaimer: I'm biased since I maintain PDK.

The advantage of this is that it is actively maintained, has an [LND reference implementation](https://github.com/chaincase-app/nolooking), and the base library supports receiving. Receiving is a huge boon for [Lightning Payjoin](https://chaincase.app/words/lightning-payjoin) fee & time savings. PDK supports coin selection and error handling so Blixt could depend on that. Since we have Serverless Payjoin on the roadmap Blixt would be able to take advantage of that technology as early as it gets merged in the library .Blixt already supports Tor so receiving via ephemeral `.onion` would be straightforward to implement from these bindings based on [Onion '78](https://github.com/armins88/onion78) or JoinMarket should the desire to receive be so great that it is accelerated on the roadmap.

The main disadvantages are that it will take some time to develop and bindings have not been tested yet.

A signal of interest from Blixt will bump the priority of a WASM bindings implementation based on what was learned developing [BDK's WASM bindings proof of concept](https://github.com/danielabrozzoni/bdk-nodejs).

## 2. The `payjoin-client` JavaScript Library

[payjoin-client](https://www.npmjs.com/package/payjoin-client) is what BlueWallet uses, and it works alright. Blixt could marry LND's PSBT protocol with the JavaScript library to send Payjoin. However, the library only supports sending and there is no active maintenance or roadmap to add receiving capability going forward

## 3. DIY roll your own.

JoinMarket has rolled their own Payjoin implementation, but its ability to negotiate fees and handle errors is limited. From what I gather having worked with the source code a number of times, it expects a happy path that creates Payjoins with exactly 2 outputs. It's so tightly coupled to the implementation that JAM's JoinMarket UI, which has been around for years, is still unable to support Payjoin.

---

Thanks again for bringing this up and planning to integrate Payjoin. I'm looking forward to hearing your opinion of these options and any I may not have thought of.

**Comment by BitcoinZavior** [comment]
This sounds amazing, I would love to have Flutter and React Native implementations of PDK. Currently we have BDK and LDK Node for Flutter and React Native, you can refer to them here: https://github.com/LtbLightning

**Comment by ConorOkus** [comment]
Hi fellas, what's the latest on this? @BitcoinZavior are there plans to have `payjoin-react-native` bindings?

**Comment by BitcoinZavior** [comment]
Yes we have PayJoin RN is on the roadmap. We will dk they once we have native iOS and Android done.

**Comment by DanGould** [comment]
I think we're still a little early but @spacebear21 has been working on JavaScript bindings for React native that seem to be working https://github.com/payjoin/rust-payjoin/pull/1190

@djkazic mentioned an interest assuming these are in fact working
